src/index.ts

```diff
@@ -31,6 +31,20 @@ export interface Options /* extends Partial<ICaBufferOpts & ICaPathOpts>  */{
   ui?: UserInterface
 }
 
+interface ICaBuffer {
+  ca: Buffer;
+}
+interface ICaPath {
+  caPath: string;
+}
+interface IDomainData {
+  key: Buffer;
+  cert: Buffer;
+}
+type IReturnCa<O extends Options> = O['getCaBuffer'] extends true ? ICaBuffer : false;
+type IReturnCaPath<O extends Options> = O['getCaPath'] extends true ? ICaPath : false;
+type IReturnData<O extends Options = {}> = (IDomainData) & (IReturnCa<O>) & (IReturnCaPath<O>);
+
```

Comment:
@zetlen Because of the additional flexibility of the API, I wrote these types that will cause the return value to be more explicit, depending on what flags you set when you call `certificateFor()`.

Let me know if you think this is too much/too complicated.  The alternative is to limit it to two different return types, but then you wind up with `Partial` properties:
```ts
type ReturnData = {
  key: Buffer,
  cert: Buffer
} | {
  ca?: Buffer,
  caPath?: string,
  key: Buffer,
  cert: Buffer
}
```

This probably wouldn't be an issue, ultimately.  I just wanted it to be more clear what you're getting for what you give, and also get the benefits of strong types.

Reply:
Thanks for the self-documenting types!

Reply:
I approve of the types; there are lot of ways to type this, but if it populates autosuggest correctly, then that's all that matters. Thanks for thinking about it carefully.